src/lib/platform.ts

```diff
@@ -24,3 +27,23 @@ function getOsName(): OsNameOptions {
 }
 
 export const osName = getOsName()
+
+export const isDesktopOrMobileApp = () => {
+  if (isElectron()) {
+    // desktop app
+    return true
+  }
+  const userAgent = window.navigator.userAgent.toLowerCase()
+  const safari = /safari/.test(userAgent)
+  const ios = /iphone|ipod|ipad/.test(userAgent)
+  if (ios && !safari) {
+    // webview(ios)
+    return true
+  }
+  const browserName = new UAParser(navigator.userAgent).getBrowser().name
+  if (browserName !== undefined && browserName.indexOf('WebView') != -1) {
+    // webview(android)
+    return true
+  }
+  return false
+}
```

Comment:
Maybe add a new line at the end of the file? This makes some code viewing/editing experiences easier.